src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Elasticsearch/V5/RequestPipeline_CallElasticsearchAsync_Integration.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.ClrProfiler.Integrations;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Elasticsearch.V5
+{
+    /// <summary>
+    /// Elasticsearch.Net.RequestPipeline.CallElasticsearchAsync&lt;T&gt; calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = ElasticsearchV5Constants.ElasticsearchAssemblyName,
+        TypeName = ElasticsearchV5Constants.RequestPipelineTypeName,
+        MethodName = "CallElasticsearchAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1<Elasticsearch.Net.ElasticsearchResponse`1<T>>",
+        ParameterTypeNames = new[] { "Elasticsearch.Net.RequestData", ClrNames.CancellationToken },
+        MinimumVersion = ElasticsearchV5Constants.Version5,
+        MaximumVersion = ElasticsearchV5Constants.Version5,
+        IntegrationName = ElasticsearchV5Constants.IntegrationName)]
+    // ReSharper disable once InconsistentNaming
+    public class RequestPipeline_CallElasticsearchAsync_Integration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TRequestData">Type of the request</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="requestData">The request data</param>
+        /// <param name="cancellationToken">The cancellation token</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TRequestData>(TTarget instance, TRequestData requestData, CancellationToken cancellationToken)
+        {
+            var scope = ElasticsearchNetCommon.CreateScope(Tracer.Instance, ElasticsearchV5Constants.IntegrationId, instance.DuckCast<RequestPipelineStruct>(), new RequestDataV5(requestData));
+
```

Comment:
Why not use a constraint for the instance, something like `IRequestPipeline` and pass to`CreateScope` just the `RequestParameters` object?

Reply:
As discussed on Slack, I'm going to keep RequestPipelineStruct because this code path is shared with the callsite instrumentation